pkg/codegen/testing/test/testdata/resource-options-pp/MyStack.java

```diff
@@ -15,7 +16,24 @@ public static void stack(Context ctx) {
             .region("us-west-2")
             .build());
 
-        var bucket1 = new Bucket("bucket1");
+        var bucket = new Bucket("bucket", BucketArgs.builder()        
+            .website(BucketWebsiteArgs.builder()
+                .indexDocument("index.html")
+                .build())
+            .build(), CustomResourceOptions.builder()
+                .provider(provider)
+                .protect(true)
+                .dependsOn(provider)
+                .build());
+
+        var bucketWithoutArgs = new Bucket("bucketWithoutArgs", BucketArgs.Empty, CustomResourceOptions.builder()
+            .provider(provider)
+            .protect(true)
+            .dependsOn(provider)
+            .ignoreChanges(            
+                bucket,
+                lifecycleRules[0])
```

Comment:
I worry this is a dangling reference. What does this refer to in the PCL file even? 

Perhaps these need to be string literals like "bucket"?

```
  @Nullable List<String> ignoreChanges,
```

Reply:
> What does this refer to in the PCL file even?

I don't know to be honest 😅 programgen doesn't typecheck PCL programs but assumes they are already valid

Reply:
Perhaps we need to ask around here, but it's plausible that PCL lets you write the string literals without quotes. In Java we do have to quote. The above code won't be right.

Reply:
You are right! expressions in `ignoreChanges` have to be stringified (asked in slack). That is, unless they are variable references 😓 that makes life extra complicated